src/core_fs/storage/workspace_storage.rs

```diff
@@ -406,9 +417,17 @@ impl WorkspaceStorage {
     fn clear_memory_cache(&self, flush: bool) -> FutureIntoCoroutine {
         let ws = self.get_storage();
 
-        FutureIntoCoroutine::from(
-            async move { ws?.clear_memory_cache(flush).await.map_err(to_py_err) },
-        )
+        FutureIntoCoroutine::from(async move {
+            let ws = ws?;
+            #[cfg(feature = "test-utils")]
+            if !flush {
+                ws.clear_cache().await;
+                return Ok(());
+            }
+            let _ = flush;
+
+            ws.flush_and_clear_memory_cache().await.map_err(to_py_err)
+        })
```

Comment:
Why do you have the logic in binding ? how do you do in rust test ?

Reply:
Before we had `clear_memory_cache(flush: bool)` which was doing 2 things:

1. Flush the cache into the storage when `flush == true`
2. Clearing the cache

In the rust test, everywhere I saw `clear_memory_cache(..)` I replaced it with `drop_memory_cache` when `flush == false` and by `flush_and_clear_memory_cache()` otherwise.

In python I can't do that because that would require a lot of change on the python code and I prefer to keep the change minimal.
On top of that, we drop (not flush) the data only during python test in this context it's OK to loose (forget) data.